### PR TITLE
Move Publish build pool back to VS 2015 machines

### PR DESCRIPTION
**Pull request by MattGal**
@brianrob please review.

### CODE REVIEW

File: buildpipeline/DotNet-Trusted-Tests-Publish.json [hunk 1]
```diff
@@ -242,18 +242,12 @@
   "quality": "definition",
```
Comment (MattGal) [inline buildpipeline/DotNet-Trusted-Tests-Publish.json]: I don't think this file's actively in use, but it shouldn't matter even if it is as publish doesn't need SourceLink.

File: BuildToolsVersion.txt [hunk 2]
```diff
@@ -1 +1 @@
-2.1.0-preview2-02516-02
```
Comment (MattGal) [inline BuildToolsVersion.txt]: This is to get MaxRetryCount defaulting back to 0, as we're not quite ready for auto-retry and that should be specified in upload-tests.proj

**Comment by brianrob** [comment]
I can't say I know much about this - this probably needs to be reviewed by @weshaggard.  That said, I appreciate the quick response.

File: buildpipeline/DotNet-Trusted-Tests-Publish.json (hunk 1 above)
Comment (weshaggard) [reply buildpipeline/DotNet-Trusted-Tests-Publish.json]: I agree I don't think this is used and likely doesn't matter.

File: buildpipeline/DotNet-Trusted-Publish.json [hunk 3]
```diff
@@ -574,17 +574,11 @@
...
-    "id": 330,
-    "name": "DotNetCore-Build",
-    "url": "https://devdiv.visualstudio.com/DefaultCollection/_apis/build/Queues/330",
+    "id": 36,
+    "name": "DotNet-Build",
     "pool": {
-      "id": 97,
-      "name": "DotNetCore-Build"
+      "id": 39,
+      "name": "DotNet-Build"
```
Comment (weshaggard) [inline buildpipeline/DotNet-Trusted-Publish.json]: While this is fine we probably should just update the definition to work with VS 2017, as I don't like the idea of multiple machine pools being necessary. That can happen independently.

File: BuildToolsVersion.txt (hunk 2 above)
Comment (weshaggard) [reply BuildToolsVersion.txt]: Is this needed to fix the corefx build currently?

File: buildpipeline/DotNet-Trusted-Publish.json (hunk 3 above)
Comment (MattGal) [reply buildpipeline/DotNet-Trusted-Publish.json]: Yes, that's my expectation, but unbreaking the builds is p0

**Comment by MattGal** [comment]
Buildtools changes look to be non-breaking and all other changes are only for official build, merging.

**Pull request merged**